lib/spack/spack/solver/concretize.lp

```diff
@@ -158,6 +158,14 @@ error(100, multiple_values_error, Attribute, Package)
      attr_single_value(Attribute),
      2 { attr(Attribute, node(ID, Package), Value) }.
 
+%-----------------------------------------------------------------------------
+% Languages used
+%-----------------------------------------------------------------------------
+
+attr("language", node(X, Package), Language) :-
+  condition_holds(ConditionID, node(X, Package)),
+  pkg_fact(Package,language(ConditionID, Language)).
```

Comment:
Is the only reason we're not modeling a language as a dependency basically because the language dependency doesn't have a well defined deptype?  I still don't see much of a difference between this and a dependency condition for a virtual.

Reply:
See https://github.com/spack/spack/pull/42062#discussion_r1521135019

Reply:
This should eventually be unnecessary once we have compilers modeled as packages.